### PR TITLE
[ci] generate historic test reports.

### PR DESCRIPTION
**Pull request by rexhoffman**
## Motivation

Generate the coverage report whenever new code lands on master.

### Have you read the [Contributing Guidelines on pull requests]

Yes

## Test Plan

Pushing to a gha-test branch.

## Related PRs

None

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

**Comment by rexhoffman** [comment]
Successfully tested here: https://github.com/diem/diem/runs/2768105254?check_suite_focus=true

File: .github/workflows/ci-update-sccache.yml [hunk 1]
```diff
@@ -54,3 +54,40 @@ jobs:
...
+      - uses: actions/checkout@v2.3.4
+        with:
+          # This ensures that the tip of the PR is checked out instead of the merge between the base ref and the tip
+          # On `push` this value will be empty and will "do-the-right-thing"
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0 #get all the history!!!
+      - uses: ./.github/actions/build-setup
+        # build setup is called to ensure the latest docker image tooling is in place and the ~/.profile is called.
+      - name: Publish unit test results
+        uses: rexhoffman/actions/process-test-results@78f868bd1170a959f0a9864e33df725690b45b9a
```
Comment (sausagee) [inline .github/workflows/ci-update-sccache.yml:80]: Point to the one in diem/actions instead?

File: .github/workflows/ci-update-sccache.yml [hunk 2]
```diff
@@ -54,3 +54,40 @@ jobs:
...
+      image: diem/build_environment:main
+      volumes:
+        - "/home/runner/work/diem/diem:/opt/git/diem"
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          # This ensures that the tip of the PR is checked out instead of the merge between the base ref and the tip
+          # On `push` this value will be empty and will "do-the-right-thing"
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0 #get all the history!!!
```
Comment (sausagee) [inline .github/workflows/ci-update-sccache.yml:76]: i think we can skip this, as this workflow runs off the main and release-* branches when they are pushed.

**Review by sausagee — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-update-sccache.yml (hunk 2 above)
Comment (rexhoffman) [reply .github/workflows/ci-update-sccache.yml:76]: We need to at least do a github pull to use the setup step in the repo.   I think it's better to be consistent in the use of the git pull.

File: .github/workflows/ci-update-sccache.yml (hunk 1 above)
Comment (rexhoffman) [reply .github/workflows/ci-update-sccache.yml:80]: ahhh yeah.... meant to change that - forgot.

**Review by sausagee — DISMISSED** [review]

**Pull request closed without merging**